sql/analyzer/apply_indexes_for_subquery_comparisons.go

```diff
@@ -31,52 +72,110 @@ import (
 // 4. The Child is a *plan.ResolvedTable.
 // 5. The referenced field in the Child is indexed.
 func applyIndexesForSubqueryComparisons(ctx *sql.Context, a *Analyzer, n sql.Node, scope *Scope, sel RuleSelector) (sql.Node, transform.TreeIdentity, error) {
-	aliases, err := getTableAliases(n, scope)
-	if err != nil {
-		return nil, transform.SameTree, err
-	}
+	var recurse func(n sql.Node, inScope *aliasScope) (sql.Node, transform.TreeIdentity, error)
+	recurse = func(n sql.Node, inScope *aliasScope) (sql.Node, transform.TreeIdentity, error) {
+		var alias, target string
+		var err error
+		switch n := n.(type) {
+		case *plan.TableAlias:
+			alias = n.Name()
+			switch c := n.Child.(type) {
+			case *plan.ResolvedTable, *plan.SubqueryAlias, *plan.ValueDerivedTable, *plan.TransformedNamedNode, *plan.RecursiveTable, *plan.DeferredAsOfTable:
+				target = c.(sql.Nameable).Name()
+			case *plan.IndexedTableAccess:
+				target = c.Name()
+			case *plan.RecursiveCte:
+			case *plan.UnresolvedTable:
+				panic("Table not resolved")
```

Comment:
Do these need to be panics? Seems like we could just return an `error` and still fail the statement. Having the entire sql-server process crash from a single bad statement sucks. I know we don't expect these to ever be hit, but seems like a good process to avoid future crashes.